### PR TITLE
Update build config to support React component lib importing by consumers

### PR DESCRIPTION
**Pull request by adamjarling**
Supports components being imported into an external React app.  

```jsx
import { ThumbnailPanel } from "@iiif/thumbnail-panel";
```

### CODE REVIEW

File: scripts/build.mjs [hunk 1]
```diff
@@ -23,6 +23,11 @@ import { execa } from 'execa';
       entry: `src/index.tsx`,
       name: 'index',
       outDir: `${DIST}/bundle`,
+      external: ['react', 'react-dom'],
```
Comment (stephenwf) [inline scripts/build.mjs:26]: May also need `react-dom/client` here too

**Review by mathewjordan — APPROVED** [review]
👍

**Comment by stephenwf** [comment]
@mathewjordan does it also work with the CodeSandbox preview?
```
npm i https://pkg.csb.dev/IIIF-Commons/thumbnail-panel/commit/68fd9158/@iiif/thumbnail-panel
```

File: package.json [hunk 2]
```diff
@@ -11,8 +11,9 @@
   ],
   "exports": {
     ".": {
+      "import": "./dist/bundle/esm/index.mjs",
       "require": "./dist/bundle/cjs/index.js",
-      "import": "./dist/bundle/esm/index.mjs"
+      "types": "./dist/index.d.ts"
```
[the remark this answers is not in the document]
Comment (stephenwf) [reply package.json]: ahh, perhaps we need:
```
"exports": {
  "." { ... },
  "./dist/index.d.ts": "./dist/index.d.ts"
}
```
Found that to be the case for css files too.

**Pull request merged**